src/io_uring/src/lib.rs

```diff
@@ -53,8 +62,11 @@ pub struct IoUring {
 }
 
 impl IoUring {
-    pub fn new(num_entries: u32) -> Result<Self> {
+    pub fn new(num_entries: u32, enable: bool) -> Result<Self> {
```

Comment:
That would work,but I prefer the more flexible approach of specifying it in a separate function.
The `IoUring` object is constructed with a Builder-like pattern, where registrations are performed with separate functions, like `register_eventfd` and `register_file` and now `register_restrictions`. I'd like to keep the consistency, WDYT?

Reply:
The builder-like pattern definitely makes sense for `register_eventfd` and `register_file`. However in the case of `register_restrictions` I think that the builder-like pattern is rather harmful because we also have to expose to the user the possibility to enable/disable the ring, which 
1. requires (small) additional knowledge from the user 
2. it's a bit harder to use
3. can be used incorrectly, resulting in runtime errors
4. the user doesn't need it apart from when registering restrictions

By receiving a slice of Restrictions as a parameter for `new()`, we would make it impossible for the customer to use the code incorrectly and easier for him to register restrictions which would be big advantages and I don't think that we would sacrifice readability by making an exception from the builder-like pattern.

Reply:
Hmm, I agree that this would make the interface easier to use and I think it would be worth it.

However, once we install the restrictions we are no longer allowed to register eventfds and other files. We would need to add new permissions in the restrictions list, which would enable users to register new files after the ring was created (essentially bypassing the `RequireFixedFds` restriction).

In order to fix this we'd have to supply all the registration fds as well as the eventfd in the constructor (which would make for a pretty cumbersome way of initialising the object, which could be fixed by adding a proper Builder pattern, a separate struct used for creating the object).

WDYT?

Reply:
I suggest to add all the registrations in the IoUring constructor. If we later think that the initialisation process is too cumbersome we may add a separate `IoUringBuilder`.
I think this is fine for now since we only create this object once (for the block device), but we should revisit this if we ever want to upstream this code to rust-vmm.

WDYT @serban300 @luminitavoicu

Reply:
I am in favor of handling all registrations inside the `IoUring` constructor, with the cost of it exposing a heavier logic, since it would be less error prone and easier to use. Add a separate `IoUringBuilder` when the need arises sounds good to me.

Reply:
Sounds good to me.

Reply:
Done, PTAL